src/runtime/runtime_data.cs

```diff
@@ -70,15 +84,10 @@ internal static void Stash()
             Marshal.WriteIntPtr(mem, (IntPtr)ms.Length);
             Marshal.Copy(data, 0, mem + IntPtr.Size, (int)ms.Length);
 
-            IntPtr capsule = PySys_GetObject("clr_data").DangerousGetAddress();
-            if (capsule != IntPtr.Zero)
-            {
-                IntPtr oldData = PyCapsule_GetPointer(capsule, null);
-                PyMem_Free(oldData);
-                PyCapsule_SetPointer(capsule, IntPtr.Zero);
-            }
-            capsule = PyCapsule_New(mem, null, IntPtr.Zero);
+            ClearCLRData();
+            IntPtr capsule = PyCapsule_New(mem, null, IntPtr.Zero);
```

Comment:
Ideally, this should have been `NewReference`, and decref below should be `Dispose`